taipy/common/config/_serializer/_base_serializer.py

```diff
@@ -155,12 +159,16 @@ def _pythonify(cls, val):
                         return _TemplateHandler._to_function(actual_val)
                     elif dynamic_type == "class":
                         return _TemplateHandler._to_class(actual_val)
+                    elif dynamic_type == "enum":
+                        return _TemplateHandler._to_enum(actual_val)
```

Comment:
Just to be sure I understand: This case does not cover internal enums (SCOPE and FREQUENCY), as they are caught up by:
```
if dynamic_type in cls._registered_types.keys():
    return cls._registered_types[dynamic_type]._pythonify(actual_val)
```
which transforms the string `SCENARIO:SCOPE` into a Scope instance with the SCENARIO value.

Your case transforms the string `Gender.male:enum` into the Gender enum with the male value.

Am I correct?

Reply:
yes, all registered types are handled first, then the enum